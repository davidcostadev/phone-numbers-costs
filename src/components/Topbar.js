import React from 'react';
import PropTypes from 'prop-types';

const Topbar = ({ title, children }) => (
  <header className="topbar">
    <h1 className="topbar__title">{title}</h1>
    <div className="topbar__right">
      {children}
    </div>
  </header>
);

Topbar.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.any.isRequired,
};

export default Topbar;
