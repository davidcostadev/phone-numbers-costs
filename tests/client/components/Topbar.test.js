import React from 'react';
import renderer from 'react-test-renderer';
import Topbar from '../../../client/components/Topbar';

describe('Topbar Component', () => {
  describe('Render', () => {
    it('default', () => {
      const tree = renderer.create((
        <Topbar title="Titulo">
          <div>children</div>
        </Topbar>
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
