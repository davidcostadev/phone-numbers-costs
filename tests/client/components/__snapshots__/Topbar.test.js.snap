// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topbar Component Render default 1`] = `
<header
  className="topbar"
>
  <h1
    className="topbar__title"
  >
    Titulo
  </h1>
  <div
    className="topbar__right"
  >
    <div>
      children
    </div>
  </div>
</header>
`;
